services/sources/publisher_backlist_service.py

```diff
@@ -96,3 +101,95 @@ def get_records_array(self,
             array_json.append(pub_backlist_records_response_json)
 
         return array_json
+    
+    def addHasPartMapping(self, resultsRecord, record):
+
+        #GOOGLE DRIVE API CALL TO GET PDF/EPUB FILES
+
+        try:
+            if 'in_copyright' in record.rights:
+                linkString = '|'.join([
+                    '1',
+                    #LINK TO PDF/EPUB,
+                    record.source,
+                    'application/pdf',
+                    '{"catalog": false, "download": true, "reader": false, "embed": false, "nypl_login": true}'
+                ])
+                record.has_part.append(linkString)
+
+            if 'public_domain' in record.rights:
+                linkString = '|'.join([
+                    '1',
+                    #LINK TO PDF/EPUB,
+                    record.source,
+                    'application/pdf',
+                    '{"catalog": false, "download": true, "reader": false, "embed": false}'
+                ])
+                record.has_part.append(linkString)
+
+        except Exception as e:
+            if e.response['Error']['Code'] == 'NoSuchKey':
+                logger.exception(PubBacklistError("Key doesn't exist"))
+            else: 
+                logger.info(PubBacklistError("Object doesn't exist"))
```

Comment:
I think we can remove the if/else here and just log the exception 

If we are unable to add the part, should we save this record?

Reply:
We shouldn't save this record if we are unable to add the part.